File: test/fixtures/java/p1/server/spring/src/main/p1/PathVariableDefinition.java
package p1;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

@RequestMapping("/")
@RestController
public class PathVariableDefinition {
    @GetMapping("/{v}")
    public void getWithNoValue(@PathVariable String v) {
    }

    @GetMapping("/{v}")
    public void getWithSingleMember(@PathVariable("v") String v) {
    }

    @GetMapping("/{v}")
    public void getWithValue(@PathVariable(value = "v") String v) {
    }

    @GetMapping("/{v}")
    public void getWithName(@PathVariable(name = "v") String v) {
    }
}

File: test/fixtures/kotlin/p1/server/spring/src/main/p1/PathVariableDefinition.kt
package p1

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RestController

@RequestMapping("/")
@RestController
class PathVariableDefinition {
    @GetMapping("/{v}")
    fun getWithNoValue(@PathVariable v: String) {
    }

    @GetMapping("/{v}")
    fun getWithSingleMember(@PathVariable("v") a: String) {
    }

    @GetMapping("/{v}")
    fun getWithValue(@PathVariable(value = "v") a: String) {
    }

    @GetMapping("/{v}")
    fun getWithName(@PathVariable(name = "v") a: String) {
    }
}
